src/k8s/pkg/k8sd/types/cluster_config_datastore.go

```diff
@@ -24,3 +29,45 @@ func (c Datastore) GetExternalClientKey() string  { return getField(c.ExternalCl
 func (c Datastore) Empty() bool {
 	return c.Type == nil && c.K8sDqlitePort == nil && c.K8sDqliteCert == nil && c.K8sDqliteKey == nil && c.ExternalURL == nil && c.ExternalCACert == nil && c.ExternalClientCert == nil && c.ExternalClientKey == nil
 }
+
+// DatastoreSnapProvider is to avoid circular dependency for snap.Snap in Datastore.ToKubeAPIServerArguments()
+type DatastorePathsProvider interface {
+	K8sDqliteStateDir() string
+	EtcdPKIDir() string
+}
+
+// ToKubeAPIServerArguments returns updateArgs, deleteArgs that can be used with snaputil.UpdateServiceArguments() for the kube-apiserver
+// according the datastore configuration.
+func (c Datastore) ToKubeAPIServerArguments(p DatastorePathsProvider) (map[string]string, []string) {
+	var (
+		updateArgs = make(map[string]string)
+		deleteArgs []string
+	)
+
+	switch c.GetType() {
+	case "k8s-dqlite":
+		updateArgs["--etcd-servers"] = fmt.Sprintf("unix://%s", path.Join(p.K8sDqliteStateDir(), "k8s-dqlite.sock"))
+		deleteArgs = []string{"--etcd-cafile", "--etcd-certfile", "--etcd-keyfile"}
+	case "external":
+		updateArgs["--etcd-servers"] = c.GetExternalURL()
+
+		// the certificates will be written by setup.EnsureExtDatastorePKI(), here we only set the paths
+		for _, loop := range []struct {
+			arg  string
+			cert string
+			path string
+		}{
+			{cert: c.GetExternalCACert(), arg: "--etcd-cafile", path: "ca.crt"},
+			{cert: c.GetExternalClientCert(), arg: "--etcd-certfile", path: "client.crt"},
+			{cert: c.GetExternalClientKey(), arg: "--etcd-keyfile", path: "client.key"},
+		} {
+			if loop.cert != "" {
```

Comment:
any need to check if the file exists first?   I see the comment above saying  that we're just setting the paths here.  Do we expect them to be written?  If the certs are updated, does something come along and restart the service (i know that's a different PR)

Reply:
good point, this is something for a separate PR (such that all `setup.Ensure*` functions return whether any of the files were changed